js/src/components/adaptive-form/adaptive-form-context.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+/**
+ * @typedef {Object} InputProps
+ * @property {*} value Form value.
+ * @property {boolean} checked Form value converted to boolean.
+ * @property {*} selected Form value.
+ * @property {(value: Event<HTMLInputElement> | *) => void} onChange Function to handle onChange event.
```

Comment:
`(value: Event<HTMLInputElement> | *)` is the same as just `(*)`

Reply:
Thanks for the suggestion. The type declaration here is mainly to make it explicitly clear that this handler is also compatible with the native `onChange` event/prop.